### PR TITLE
triage connector v1.0.0

### PR DESCRIPTION
**Pull request by cs-omkar-kumbhar**
#### Descriptions:
PMDB Link: https://pmdb.fortinet.com/ProjectManagement/editprojects.php?id=22909

UTCs:
Connector installation verified.
Connector logo verified.
Playbook list verified.
Playbook tags verified.
All Playbook are in info mode verified.
All Playbook are in inactive mode verified.
Verified all actions.

### CODE REVIEW

File: hatching-triage/connector.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+logger = get_logger("hatching-triage")
+
+
+class HatchingTriage(Connector):
+    def execute(self, config, operation, params, **kwargs):
+        try:
+            logger.error(f"----------------------operation1: {operation}")
+            operation = operations.get(operation)
+            logger.error(f"----------------------operation2: {operation}")
```
Comment (cs-harsh-sharma) [inline hatching-triage/connector.py]: why is this logger.error twice for??

File: hatching-triage/info.json [hunk 2]
```diff
@@ -0,0 +1,857 @@
+{
+  "name": "hatching-triage",
+  "version": "1.0.0",
+  "label": "Hatching Triage",
+  "category": "Threat Intelligence",
```
Comment (cs-harsh-sharma) [inline hatching-triage/info.json]: Can you confirm connector category once with prerna

File: hatching-triage/info.json [hunk 3]
```diff
@@ -0,0 +1,857 @@
...
+            {
+              "profile": "1111a5f4-dd72-4551-adc4-2165626156de"
+            }
+          ],
+          "required": false,
+          "editable": true,
+          "visible": true
+        }
+      ],
+      "output_schema": {}
```
Comment (cs-harsh-sharma) [inline hatching-triage/info.json:376]: no output schema?

File: hatching-triage/info.json [hunk 4]
```diff
@@ -0,0 +1,857 @@
...
+            "sim200",
+            "sim404",
+            "simnx"
+          ],
+          "visible": true,
+          "editable": true,
+          "required": false
+        }
+      ],
+      "output_schema": {}
```
Comment (cs-harsh-sharma) [inline hatching-triage/info.json:776]: missing output schema

File: hatching-triage/info.json [hunk 5]
```diff
@@ -0,0 +1,857 @@
...
+          "name": "profile_id",
+          "tooltip": "Specify the profile ID of the profile that you want to delete.",
+          "description": "Specify the profile ID of the profile that you want to delete.",
+          "type": "text",
+          "visible": true,
+          "editable": true,
+          "required": true
+        }
+      ],
+      "output_schema": {}
```
Comment (cs-harsh-sharma) [inline hatching-triage/info.json:797]: no output schema

File: hatching-triage/operations.py [hunk 6]
```diff
@@ -0,0 +1,241 @@
+""" Copyright start
+  Copyright (C) 2008 - 2023 Fortinet Inc.
+  All rights reserved.
+  FORTINET CONFIDENTIAL & FORTINET PROPRIETARY SOURCE CODE
+  Copyright end """
+import json
```
Comment (cs-harsh-sharma) [inline hatching-triage/operations.py:7]: little reformatting req.

File: hatching-triage/operations.py [hunk 7]
```diff
@@ -0,0 +1,241 @@
...
+    def api_request(self, method, endpoint, params={}, data={}, files=None):
+        try:
+            endpoint = self.url + endpoint
+            headers = {"Authorization": f"Bearer {self.api_key}"}
+            logger.error(f"----------------------method: {method} | endpoint: {endpoint} | headers: {headers} | params: {params} | data: {data}")
+            response = request(method, endpoint, headers=headers, params=params, data=data, files=files, verify=self.verify_ssl)
+
+            if response.status_code in [200, 201, 204]:
+                if response.text != "":
+                    j = response.json()
```
Comment (cs-harsh-sharma) [inline hatching-triage/operations.py]: parameter "j" can have a better meaning full name

File: hatching-triage/info.json (hunk 3 above)
Comment (cs-omkar-kumbhar) [reply hatching-triage/info.json:376]: the output schema that we get is like this only. We just get the empty dict.

File: hatching-triage/info.json (hunk 4 above)
Comment (cs-omkar-kumbhar) [reply hatching-triage/info.json:776]: the output schema that we get is like this only. We just get the empty dict.

File: hatching-triage/info.json (hunk 5 above)
Comment (cs-omkar-kumbhar) [reply hatching-triage/info.json:797]: the output schema that we get is like this only. We just get the empty dict.

File: hatching-triage/operations.py (hunk 6 above)
Comment (cs-omkar-kumbhar) [reply hatching-triage/operations.py:7]: done

File: hatching-triage/connector.py (hunk 1 above)
Comment (cs-omkar-kumbhar) [reply hatching-triage/connector.py]: done

File: hatching-triage/operations.py (hunk 7 above)
Comment (cs-omkar-kumbhar) [reply hatching-triage/operations.py]: done

File: hatching-triage/info.json (hunk 2 above)
Comment (cs-omkar-kumbhar) [reply hatching-triage/info.json]: @Prerna this is a file and url analysis tool, so 'Threat Intelligence' should be the connector category, right ?

Comment (cs-prerna-joshi) [reply hatching-triage/info.json]: Category would be **Malware Analysis**

File: hatching-triage/info.json [hunk 8]
```diff
@@ -0,0 +1,857 @@
...
+          "submitted": ""
+        }
+      }
+    },
+    {
+      "operation": "get_sample",
+      "description": "Retrieves the sample details based on the sample ID that you have specified.",
+      "category": "investigation",
+      "annotation": "get_sample",
+      "title": "Get Sample",
```
Comment (cs-prerna-joshi) [inline hatching-triage/info.json]:
```suggestion
      "title": "Get Sample by ID",
```

File: hatching-triage/info.json [hunk 9]
```diff
@@ -0,0 +1,857 @@
...
+            "resource":"",
+            "platform":"",
+            "queue_id":""
+          }
+        }
+      }
+    },
+    {
+      "operation": "set_sample_profile",
+      "description": "When a sample is in the static_analysis status, a profile should be selected in order to continue.",
```
Comment (cs-prerna-joshi) [inline hatching-triage/info.json]: update description

File: hatching-triage/info.json [hunk 10]
```diff
@@ -0,0 +1,857 @@
...
+        "unpack_count": "",
+        "error_count": ""
+      }
+    },
+    {
+      "operation": "get_report_triage",
+      "description": "Retrieves the generated Triage Report for a single task based on the sample ID and task ID that you have provided.",
+      "category": "investigation",
+      "annotation": "get_report_triage",
+      "title": "Get Report Triage",
```
Comment (cs-prerna-joshi) [inline hatching-triage/info.json]:
```suggestion
      "title": "Get Triage Report",
```

**Review by cs-prerna-joshi — APPROVED** [review]

**Pull request merged**